runtimes/protocol/lsp.ts

```diff
@@ -40,6 +41,50 @@ export class AutoParameterStructuresProtocolRequestType<P, R, PR, E, RO>
     }
 }
 
+/**
+ * Custom Client extension information passed during initialization handshake from Client to Server.
+ */
+export interface AWSClientInfoInitializationOptions {
+    /**
+     * Client extension name, may contain spaces.
+     */
+    name: string
+    version: string
+    /**
+     * Custom Client ID value, set based on client extension internal business logic.
+     * Can be used to uniquely identify installations of Client extension.
+     */
+    clientId?: string
+}
+
+/**
+ * Information about a Platform or environment.
+ * Use it to pass information about IDE or text editor, in which Client extension is running.
+ */
+export interface AWSPlatformInitializationOptions {
+    /**
+     * Platform name, may contain spaces.
+     */
+    name: string
+    version: string
+}
+
+export interface AWSInitializationOptions {
```

Comment:
Do I understand correctly that we won't have 'serverInfo' here? so we won't know the name and version of the server

Reply:
`serverInfo` is known to Runtime, it is injected with RuntimeProps in build configuration